File: compatLib/compatLibVU/src/main/java/app/lawnchair/compatlib/fourteen/ActivityOptionsCompatVU.java
package app.lawnchair.compatlib.fourteen;

import android.app.ActivityOptions;
import android.content.Context;
import android.os.Handler;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;
import app.lawnchair.compatlib.thirteen.ActivityOptionsCompatVT;

@RequiresApi(34)
public class ActivityOptionsCompatVU extends ActivityOptionsCompatVT {
    @NonNull
    @Override
    public ActivityOptions makeCustomAnimation(
            @NonNull Context context,
            int enterResId,
            int exitResId,
            @NonNull final Handler callbackHandler,
            @Nullable final Runnable startedListener,
            @Nullable final Runnable finishedListener) {
        return ActivityOptions.makeCustomAnimation(
                context,
                enterResId,
                exitResId,
                0,
                callbackHandler,
                new ActivityOptions.OnAnimationStartedListener() {
                    @Override
                    public void onAnimationStarted(long elapsedRealTime) {
                        if (startedListener != null) {
                            startedListener.run();
                        }
                    }
                },
                new ActivityOptions.OnAnimationFinishedListener() {
                    @Override
                    public void onAnimationFinished(long elapsedRealTime) {
                        if (finishedListener != null) {
                            finishedListener.run();
                        }
                    }
                });
    }
}

File: compatLib/compatLibVU/src/app/lawnchair/compatlib/fourteen/ActivityOptionsCompatVU.kt
package app.lawnchair.compatlib.fourteen;

import android.app.ActivityOptions;
import android.content.Context;
import android.os.Handler;
import android.view.RemoteAnimationAdapter;
import android.window.RemoteTransition;

import app.lawnchair.compatlib.ActivityOptionsCompat;

class ActivityOptionsCompatVU : ActivityOptionsCompat() {
    override fun makeCustomAnimation(
        context: Context,
        enterResId: Int,
        exitResId: Int,
        callback: Runnable?,
        callbackHandler: Handler?
    ): ActivityOptions {
       TODO()
    }

    override fun makeRemoteAnimation(
        remoteAnimationAdapter: RemoteAnimationAdapter,
        remoteTransition: Any,
        debugName: String
    ): ActivityOptions {
        return ActivityOptions.makeRemoteAnimation(
            remoteAnimationAdapter,
            remoteTransition as RemoteTransition
        )
    }
}
